src/modules/core/components/Fields/Toggle/Toggle.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { useField } from 'formik';
+import { PopperProps } from 'react-popper';
+
+import InputLabel from '~core/Fields/InputLabel';
+import Icon from '~core/Icon';
+import { Tooltip } from '~core/Popover';
+
+import styles from './Toggle.css';
+
+const displayName = 'Toggle';
+
+interface Props {
+  name: string;
+  label?: string;
+  disabled?: boolean;
+  tooltipText?: string;
+  elementOnly?: boolean;
+  /** Options to pass through the <Popper> element. See here: https://github.com/FezVrasta/react-popper#api-documentation */
+  tooltipPopperProps?: Omit<PopperProps, 'children'>;
+}
+
+const Toggle = ({
+  name,
+  label,
+  disabled = false,
+  elementOnly = false,
+  tooltipText,
+  tooltipPopperProps = {
+    placement: 'right-start',
+    modifiers: [
+      {
+        name: 'offset',
+        options: {
+          offset: [-3, 10],
+        },
+      },
+    ],
+  },
+}: Props) => {
+  const [{ onChange, value }] = useField(name);
+
+  return (
+    <div className={styles.container}>
+      {!elementOnly && label && (
+        <InputLabel label={label} appearance={{ colorSchema: 'grey' }} />
+      )}
+      <div>
+        <input
+          name={name}
+          type="checkbox"
+          disabled={disabled}
+          aria-checked={value}
+          aria-disabled={disabled}
+          className={styles.delegate}
+          onChange={onChange}
```

Comment:
You might need to create a custom method for this.

I've noticed a bug.

If the initial value is `true` and you try to click it, it takes to clicks to change the state:
- 1st click does nothing
- 2nd click changes the state to `false`

This only happens the first time around. All subsequent clicks work as expected.

I wasn't able to pinpoint the cause of this so you'll need to do some investigating. _(it works as expected if the initial value is `false`)_

Might be because the `toggleSwitch` is positioned absolute, but again, I'm not sure.

Here's the setup I used to test:
```js
      <Form onSubmit={console.log} initialValues={{ toggle: true }}>
        <Toggle name="toggle" />
        <Button type="submit" text={{ id: 'button.confirm' }} />
      </Form>
```